scrapy/linkextractors/lxmlhtml.py

```diff
@@ -66,7 +66,7 @@ def _extract_links(self, selector, response_url, response_encoding, base_url):
                 url = self.process_attr(attr_val)
                 if url is None:
                     continue
-            url = to_unicode(url, encoding=response_encoding)
+            url = canonicalize_url(url, encoding=response_encoding)
```

Comment:
I'm not sure this is correct; canonicalize_url is not guaranteed to be safe - it could be the case that server won't accept URL after canonicalization. We should never be downloading results of canonicalize_url; its use should be limited to duplicate checks.